tests/functional/api/test_merge_requests.py

```diff
@@ -181,11 +181,27 @@ def test_merge_request_reset_approvals(gitlab_url, project, wait_for_sidekiq):
     assert mr.reset_approvals()
 
 
+def test_cancel_merge_when_pipeline_succeeds(project, merge_request, wait_for_sidekiq):
+    mr = merge_request(source_branch="test_merge_request_merge", create_pipeline=True)
+    # Set to merge when the pipeline succeeds, which should never happen
+    mr.merge(merge_when_pipeline_succeeds=True)
+    wait_for_sidekiq(timeout=60)
+
+    mr = project.mergerequests.get(mr.iid)
+    assert mr.merged_at is None
+    assert mr.merge_when_pipeline_succeeds is True
```

Comment:
just a nitpicky question here if we should follow pep8 as we are dealing with a boolean :)

Reply:
Not sure I understand.

Do you mean:
`assert mr.merge_when_pipeline_succeeds` ?

But that would pass if it was a 1 or a non-empty string or pretty much any value that is not False, 0, or an empty string.

Reply:
Yes, my understanding of the pep is it recommends against this when comparing against a known boolean type, and here the API explicitly returns a boolean. But maybe we shouldn't waste time on this :)